### PR TITLE
MGMT-9095: Opeshift versions sorting order should be new --> old

### PR DESCRIPTION
**Pull request by batzionb**

### CODE REVIEW

**Comment by batzionb** [comment]
Fixed sorting to be alpha-numeric

File: src/ocm/api/axiosClient.ts [hunk 1]
```diff
@@ -15,17 +15,21 @@ const axiosCaseConverterOptions = {
...
-  return applyCaseMiddleware(client, axiosCaseConverterOptions);
+  if (applyCase) {
+    return applyCaseMiddleware(client, axiosCaseConverterOptions);
+  }
+  return client;
 };
 
 let client: AxiosInstance = getDefaultClient();
 let ocmClient: AxiosInstance | null;
+let noCoverterClient = getDefaultClient(false);
```
Comment (jkilzi) [inline src/ocm/api/axiosClient.ts:32]: `noCoverter` ?

File: src/ocm/api/axiosClient.ts [hunk 2]
```diff
@@ -41,6 +45,7 @@ export const setAuthInterceptor = (authInterceptor: (client: AxiosInstance) => A
     aiInterceptor(authInterceptor(axios.create())),
     axiosCaseConverterOptions,
   );
+  noCoverterClient = aiInterceptor(authInterceptor(axios.create()));
```
Comment (jkilzi) [inline src/ocm/api/axiosClient.ts:48]: `SetAuthInterceptor` is called by uhc-portal, the whole authentication handling is performed by them. Di you test this change didn't break anything in the uhc-portal?

File: src/ocm/services/apis/SupportedOpenshiftVersionsAPI.ts [hunk 3]
```diff
@@ -7,7 +7,8 @@ const SupportedOpenshiftVersionsAPI = {
   },
 
   list() {
-    return client.get<OpenshiftVersion>(`${SupportedOpenshiftVersionsAPI.makeBaseURI()}`);
+    //use client without camel case converter, since openshift verion keys can contain hyphens
+    return noCoverterClient.get<OpenshiftVersion>(`${SupportedOpenshiftVersionsAPI.makeBaseURI()}`);
```
Comment (jkilzi) [inline src/ocm/services/apis/SupportedOpenshiftVersionsAPI.ts:11]: Can't we just tweak the `axios-case-converter`?

**Review by jkilzi — CHANGES_REQUESTED** [review]

File: src/ocm/hooks/useOpenshiftVersions.tsx [hunk 4]
```diff
@@ -48,12 +52,13 @@ export default function useOpenshiftVersions(): UseOpenshiftVersionsType {
...
-      version,
-    [versions],
-  );
+  const normalizeClusterVersion = React.useCallback((version = '') => {
+    if (!version) {
+      return '';
+    }
+    const [major, minor] = version.split('.');
+    return `${major}.${minor}`;
+  }, []);
```
Comment (jkilzi) [inline src/ocm/hooks/useOpenshiftVersions.tsx:61]: how this change affects day2?

**Pull request closed without merging**